src/main/scala/intellij/haskell/external/component/PathComponent.scala

```diff
@@ -0,0 +1,119 @@
+package intellij.haskell.external.component
+
+import java.io.File
+import java.net.URL
+
+import com.intellij.openapi.project.Project
+import com.intellij.util.io.URLUtil
+import intellij.haskell.HaskellNotificationGroup
+import intellij.haskell.external.commandLine.StackCommandLine
+import org.apache.commons.io.FileUtils
+
+import scala.io.Source
+
+object PathComponent {
+  private final val PackageNamePattern = """.* (.*) [==|installed].*""".r
+  private final val LocalPkgdbResolverPattern = """.*/(lts-\d+\.\d+)/.*\n""".r
+  private final val CabalConfigFileResolverPattern = """.*/(lts-\d+\.\d+)""".r
+
+  def getIdeaPath(project: Project): String = {
+    project.getBasePath + File.separator + ".idea"
+  }
+
+  def getConfigFilePath(project: Project): String = {
+    getIdeaPath(project) + File.separator + "cabal.config"
+  }
+
+  def getAllAvailablePackageNames(project: Project): Option[Iterable[String]] = {
+    val configFilePath = getConfigFilePath(project)
+    val configFile = new File(configFilePath)
+    if (!configFile.exists()) {
+      downloadAndParseCabalConfigFile(project)
+    } else {
+      val needUpdate = for {
+        oldResolver <- getResolverFromCabalConfigFile(project)
+        newResolver <- getResolverFromLocalPkgdb(project)
+      } yield oldResolver != newResolver
```

Comment:
the old resolver is read from the existed cabal.config file and the new resolver is set by users, if users change the Stack resolver then `oldResolver != newResolver` will be true, just to make sure the cabal.config file can match up with the Stack resolver in the project.